etna/pipeline/hierarchical_pipeline.py

```diff
@@ -0,0 +1,156 @@
+from copy import deepcopy
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Sequence
+
+from etna.datasets.tsdataset import TSDataset
+from etna.datasets.utils import get_target_with_quantiles
+from etna.loggers import tslogger
+from etna.metrics import MAE
+from etna.metrics import Metric
+from etna.models.base import ModelType
+from etna.pipeline.pipeline import Pipeline
+from etna.reconciliation.base import BaseReconciliator
+from etna.transforms.base import Transform
+
+
+class HierarchicalPipeline(Pipeline):
```

Comment:
After all, it is somewhat confusing that `HierarchicalPipeline` mimics `Pipeline`, but we can't e.g. mimic the logic of `AutoRegressivePipeline` or predict lower levels with ensembles.

May be it should be decorator over pipeline class that only adds logic with reconciliation.

Reply:
Have we thought about this possibility?

Reply:
We thought about solving it via composition, however it turns out that we should implement too much methods. But it is interesting point that levels might be predicted via different pipelines type, we haven't thought in this direction. May be we will improve this in next iteration of this track

Reply:
We should definitely think about it in the future.